hoomd/md/compute.py

```diff
@@ -0,0 +1,276 @@
+        self._cpp_obj = thermo_cls(self._simulation.state._cpp_sys_def, group, "")
+        super()._attach()
+
+    @log
+    def kinetic_temperature(self):
+        R"""
+        :math:`kT`, instantaneous thermal energy of the group (in energy units).
+
+        Calculated as
+
+          .. math::
+
+            kT = 2 \cdot \frac{K}{N_{\mathrm{dof}}}
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.kinetic_temperature
+        else:
+            return None
+
+    @log
+    def pressure(self):
+        R"""
+        :math:`P`, instantaneous pressure of the group (in pressure units).
+
+        Calculated as
+
+          .. math::
+
+              W = \frac{1}{2} \sum_{i}\sum_{j \ne i} \vec{F}_{ij} \cdot \vec{r_{ij}} + \sum_{k} \vec{F}_{k} \cdot \vec{r_{k}}
+
+          where :math:`\vec{F}_{ij}` are pairwise forces between particles and :math:`\vec{F}_k` are forces due to explicit constraints, implicit rigid
+          body constraints, external walls, and fields. In 2D simulations,
+          :math:`P = (K + \frac{1}{2}\cdot W)/A` where :math:`A` is the area of the simulation box.
+          of the simulation box.
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.pressure
+        else:
+            return None
+
+    @log(flag='sequence')
+    def pressure_tensor(self):
+        R"""
+        (:math:`P_{xx}`, :math:`P_{xy}`, :math:`P_{xz}`, :math:`P_{yy}`, :math:`P_{yz}`, :math:`P_{zz}`).
+
+        Instantaneous pressure tensor of the group (in pressure units).
+
+          .. math::
+
+              P_{ij} = \left[  \sum_{k\in[0..N)} m_k v_{k,i} v_{k,j} +
+                               \sum_{k\in[0..N)} \sum_{l > k} \frac{1}{2} \left(\vec{r}_{kl,i} \vec{F}_{kl,j} + \vec{r}_{kl,j} \vec{F}_{kl, i} \right) \right]/V
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.pressure_tensor
+        else:
+            return None
+
+    @log
+    def kinetic_energy(self):
+        R""" :math:`K`, total kinetic energy of all particles in the group (in energy units). """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.kinetic_energy
+        else:
+            return None
+
+    @log
+    def translational_kinetic_energy(self):
+        R""" :math:`K_{\mathrm{trans}}`, translational kinetic energy of all particles in the group (in energy units). """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.translational_kinetic_energy
+        else:
+            return None
+
+    @log
+    def rotational_kinetic_energy(self):
+        R""" :math:`K_{\mathrm{rot}}`, rotational kinetic energy of all particles in the group (in energy units). """
```

Comment:
Add a formula to make the computation explicit.

Reply:
I'm trying to look through the source code to figure out the formula myself, can you clarify what exactly is happening around line 280 of `md/ComputeThermo.cc`? I don't understand what the variable `s` is or why `p` is a quaternion.

Reply:
This is `1/2(L_x**2/I_x + L_y**2 / I_y+ L_z**2 / I_z)` where I is the moment of inertia and L is the angular momentum in the (diagonal) reference frame of the particle.